src/lib.rs

```diff
@@ -480,7 +458,7 @@ pub trait KeystoreBackend<'a, L: Ledger>: Send {
 }
 
 /// Transient state derived from the persistent [KeystoreState].
-pub struct KeystoreSession<
+pub struct KeystoreModel<
```

Comment:
So having looked all this over I am thinking it makes more sense to pull out KeystoreModel from KeystoreSession.  I think the `KeystoreMdoel` should represent the storage layer and the `KeystoreSession` seems like it should be sort of the working state of the keystore (including pointers to storage).  I think the plan is to eventually have `KeystoreModel` replace `KeystoreState`

What are your thoughts on having something like
```
struct KeystoreModel {
    assets: Assets,
    transactions: Transactions
    ...
}
```
And having all the logic you put in `create_stores` used to construct the it.

Reply:
I tried adding `KeystoreModel` and constructing the storage using this but realized that since `KeystoreModel` has an `Arc<Mutex<AtomicKeystoreStorage<'a, L, Meta>>>` field for the persistence field, whereas the `create_stores` function as well as functions and tests that call `create_stores` actually use `AtomicKeystoreStorage` without `Arc<Mutex>>`, we can't really simplify `create_stores` much. I'm leaning towards not pulling out the storage layer from the other stuff now.

Reply:
But I'll rename the `storage` field of `KeystoreModel` to `persistence` to avoid confusion, since now `assets`, `persistence` and `atomic_store` are all storage.

Reply:
Ok lets then follow up with this after, can you create an issue for refactoring out all the storage stuff into it's own struct?

Reply:
Sure! I just created an issue: https://github.com/EspressoSystems/seahorse/issues/227.